diff -up wine-5.21/dlls/gdi32/dibdrv/opengl.c.civ3~ wine-5.21/dlls/gdi32/dibdrv/opengl.c
--- wine-5.21/dlls/gdi32/dibdrv/opengl.c.civ3~	2020-11-09 00:40:02.667706283 +0100
+++ wine-5.21/dlls/gdi32/dibdrv/opengl.c	2020-11-09 14:05:18.623783559 +0100
@@ -192,7 +192,7 @@ static BOOL CDECL osmesa_make_current( s
         return TRUE;
     }
 
-    type = context->format == OSMESA_RGB_565 ? GL_UNSIGNED_SHORT_5_6_5 : GL_UNSIGNED_BYTE;
+    type = GL_UNSIGNED_BYTE; // This is more accurate: context->format == OSMESA_RGB_565 ? GL_UNSIGNED_SHORT_5_6_5 : GL_UNSIGNED_BYTE; - but it breaks Civ3
     ret = pOSMesaMakeCurrent( context->context, bits, type, width, height );
     if (ret)
     {
