diff -up wine-6.12/patches/patchinstall.sh.omv~ wine-6.12/patches/patchinstall.sh
--- wine-6.12/patches/patchinstall.sh.omv~	2021-07-04 02:15:53.827651165 +0200
+++ wine-6.12/patches/patchinstall.sh	2021-07-04 02:16:09.871244312 +0200
@@ -4204,18 +4204,6 @@ if test "$enable_wintrust_WTHelperGetPro
 	patch_apply wintrust-WTHelperGetProvCertFromChain/0001-wintrust-Add-parameter-check-in-WTHelperGetProvCertF.patch
 fi
 
-# Patchset ws2_32-connect-already-connected
-# |
-# | This patchset fixes the following Wine bugs:
-# |   *	[#51381] ws2_32: return correct error code when already connected.
-# |
-# | Modified files:
-# |   *	server/sock.c
-# |
-if test "$enable_ws2_32_connect_already_connected" -eq 1; then
-	patch_apply ws2_32-connect-already-connected/0001-server-Explicitly-forbid-connecting-a-listening-or-c.patch
-fi
-
 # Patchset wscript-support-d-u-switches
 # |
 # | This patchset fixes the following Wine bugs:
