diff -up wine-9.12/dlls/ntdll/unix/esync.c.omv~ wine-9.12/dlls/ntdll/unix/esync.c
--- wine-9.12/dlls/ntdll/unix/esync.c.omv~	2024-06-30 13:43:48.391613989 +0200
+++ wine-9.12/dlls/ntdll/unix/esync.c	2024-06-30 14:59:39.631717479 +0200
@@ -69,6 +69,12 @@ int do_esync(void)
 #endif
 }
 
+#ifdef __i386__
+typedef long inttype;
+#else
+typedef int inttype;
+#endif
+
 struct esync
 {
     enum esync_type type;
@@ -79,7 +85,7 @@ struct esync
 struct semaphore
 {
     int max;
-    int count;
+    LONG count;
 };
 C_ASSERT(sizeof(struct semaphore) == 8);
 
@@ -92,8 +98,8 @@ C_ASSERT(sizeof(struct mutex) == 8);
 
 struct event
 {
-    int signaled;
-    int locked;
+    LONG signaled;
+    LONG locked;
 };
 C_ASSERT(sizeof(struct event) == 8);
 
@@ -180,7 +186,7 @@ static struct esync *add_to_list( HANDLE
         }
     }
 
-    if (!InterlockedCompareExchange( (int *)&esync_list[entry][idx].type, type, 0 ))
+    if (!InterlockedCompareExchange( (LONG *)&esync_list[entry][idx].type, type, 0 ))
     {
         esync_list[entry][idx].fd = fd;
         esync_list[entry][idx].shm = shm;
@@ -272,7 +278,7 @@ NTSTATUS esync_close( HANDLE handle )
 
     if (entry < ESYNC_LIST_ENTRIES && esync_list[entry])
     {
-        if (InterlockedExchange((int *)&esync_list[entry][idx].type, 0))
+        if (InterlockedExchange((LONG *)&esync_list[entry][idx].type, 0))
         {
             close( esync_list[entry][idx].fd );
             return STATUS_SUCCESS;
