diff -up wine-5.7/dlls/schedsvc/tests/schrpc_c.c.gcc10~ wine-5.7/dlls/schedsvc/tests/schrpc_c.c
--- wine-5.7/dlls/schedsvc/tests/schrpc_c.c.gcc10~	2020-05-06 21:26:10.246680129 +0200
+++ wine-5.7/dlls/schedsvc/tests/schrpc_c.c	2020-05-06 21:43:06.606424681 +0200
@@ -148,7 +148,7 @@ static const MIDL_PROC_FORMAT_STRING __M
  * ITaskSchedulerService interface
  */
 
-handle_t schrpc_handle;
+extern handle_t schrpc_handle;
 
 static const RPC_CLIENT_INTERFACE ITaskSchedulerService___RpcClientInterface =
 {
