diff -up wine-9.12/loader/main.c.1~ wine-9.12/loader/main.c
--- wine-9.12/loader/main.c.1~	2024-06-30 12:58:52.510299774 +0200
+++ wine-9.12/loader/main.c	2024-06-30 13:00:14.967153888 +0200
@@ -256,6 +256,12 @@ int main( int argc, char *argv[] )
 {
     void *handle;
 
+    /* Workaround for crackling audio with many games (e.g. Civ3, Broken Sword 4, ...) */
+    if (!getenv( "PULSE_LATENCY_MSEC" ) && !getenv( "WINE_DONT_MESS_WITH_PULSE" ))
+    {
+        setenv( "PULSE_LATENCY_MSEC", "60", 1);
+    }
+
     init_reserved_areas();
 
     if ((handle = load_ntdll( argv[0] )))
