diff -up wine-5.6/dlls/gdi32/freetype.c.omv~ wine-5.6/dlls/gdi32/freetype.c
--- wine-5.6/dlls/gdi32/freetype.c.omv~	2020-04-11 15:38:12.835590112 +0200
+++ wine-5.6/dlls/gdi32/freetype.c	2020-04-11 15:38:33.502829989 +0200
@@ -2790,7 +2790,7 @@ static UINT parse_aa_pattern( FcPattern
 
 static void init_fontconfig(void)
 {
-    void *fc_handle = dlopen(SONAME_LIBFONTCONFIG, RTLD_NOW);
+    void *fc_handle = dlopen(SONAME_LIBFONTCONFIG, RTLD_LAZY);
 
     if (!fc_handle)
     {
@@ -4160,7 +4160,7 @@ static void update_font_info(void)
 
 static BOOL init_freetype(void)
 {
-    ft_handle = dlopen(SONAME_LIBFREETYPE, RTLD_NOW);
+    ft_handle = dlopen(SONAME_LIBFREETYPE, RTLD_LAZY);
     if(!ft_handle) {
         WINE_MESSAGE(
       "Wine cannot find the FreeType font library.  To enable Wine to\n"
diff -up wine-5.6/loader/main.c.omv~ wine-5.6/loader/main.c
