diff -up wine-9.9/dlls/shell32/iconcache.c.omv~ wine-9.9/dlls/shell32/iconcache.c
--- wine-9.9/dlls/shell32/iconcache.c.omv~	2024-05-18 17:49:15.037486427 +0200
+++ wine-9.9/dlls/shell32/iconcache.c	2024-05-18 17:51:00.494647720 +0200
@@ -351,7 +351,9 @@ static INT SIC_LoadIcon (const WCHAR *so
     /* Keep track of the sizes in case any icon fails to get extracted */
     for (i = 0; i < ARRAY_SIZE(hicons); i++)
     {
-        ImageList_GetIconSize(shell_imagelists[i], &size[i].cx, &size[i].cy);
+        INT cx, cy;
+        ImageList_GetIconSize(shell_imagelists[i], &cx, &cy);
+	size[i].cx = cx; size[i].cy = cy;
         PrivateExtractIconsW(sourcefile, index, size[i].cx, size[i].cy, &hicons[i], 0, 1, 0);
     }
 
diff -up wine-9.9/dlls/ntdll/unix/esync.c.omv~ wine-9.9/dlls/ntdll/unix/esync.c
--- wine-9.9/dlls/ntdll/unix/esync.c.omv~	2024-05-18 18:11:04.039513464 +0200
+++ wine-9.9/dlls/ntdll/unix/esync.c	2024-05-18 18:15:38.090467956 +0200
@@ -79,7 +79,7 @@ struct esync
 struct semaphore
 {
     int max;
-    int count;
+    LONG count;
 };
 C_ASSERT(sizeof(struct semaphore) == 8);
 
@@ -92,8 +92,8 @@ C_ASSERT(sizeof(struct mutex) == 8);
 
 struct event
 {
-    int signaled;
-    int locked;
+    LONG signaled;
+    LONG locked;
 };
 C_ASSERT(sizeof(struct event) == 8);
 
@@ -180,7 +180,7 @@ static struct esync *add_to_list( HANDLE
         }
     }
 
-    if (!InterlockedCompareExchange( (int *)&esync_list[entry][idx].type, type, 0 ))
+    if (!InterlockedCompareExchange( (LONG *)&esync_list[entry][idx].type, type, 0 ))
     {
         esync_list[entry][idx].fd = fd;
         esync_list[entry][idx].shm = shm;
@@ -272,7 +272,7 @@ NTSTATUS esync_close( HANDLE handle )
 
     if (entry < ESYNC_LIST_ENTRIES && esync_list[entry])
     {
-        if (InterlockedExchange((int *)&esync_list[entry][idx].type, 0))
+        if (InterlockedExchange((LONG *)&esync_list[entry][idx].type, 0))
         {
             close( esync_list[entry][idx].fd );
             return STATUS_SUCCESS;
diff -up wine-9.9/dlls/win32u/dibdrv/graphics.c.omv~ wine-9.9/dlls/win32u/dibdrv/graphics.c
--- wine-9.9/dlls/win32u/dibdrv/graphics.c.omv~	2024-05-18 19:02:47.043275459 +0200
+++ wine-9.9/dlls/win32u/dibdrv/graphics.c	2024-05-18 19:03:23.271688380 +0200
@@ -1538,7 +1538,8 @@ BOOL dibdrv_RoundRect( PHYSDEV dev, INT
     DC *dc = get_physdev_dc( dev );
     RECT rect;
     POINT pt[2], *points;
-    int i, end, count;
+    int i, end;
+    ULONG count;
     BOOL ret = TRUE;
     HRGN outline = 0, interior = 0;
 
@@ -1632,7 +1633,7 @@ BOOL dibdrv_RoundRect( PHYSDEV dev, INT
     }
 
     if (pdev->brush.style != BS_NULL &&
-        !(interior = ULongToHandle(NtGdiPolyPolyDraw( ULongToHandle(ALTERNATE), points, (const UINT *)&count, 1, NtGdiPolyPolygonRgn ))))
+        !(interior = ULongToHandle(NtGdiPolyPolyDraw( ULongToHandle(ALTERNATE), points, &count, 1, NtGdiPolyPolygonRgn ))))
     {
         free( points );
         if (outline) NtGdiDeleteObjectApp( outline );
