Index: wine-1.0-rc1/dlls/ntdll/server.c
===================================================================
--- wine-1.0-rc1/dlls/ntdll/server.c
+++ wine-1.0-rc1/dlls/ntdll/server.c	2008-05-13 23:02:38.000000000 +0300
@@ -772,6 +772,11 @@
     mkdir( "drive_c", 0777 );
     symlink( "../drive_c", "dosdevices/c:" );
     symlink( "/", "dosdevices/z:" );
+
+    /* OpenMandriva specific */
+    symlink( "/media/floppy", "dosdevices/a:" );
+    symlink( getenv( "HOME" ), "dosdevices/d:" );
+    symlink( "/dev/ttyUSB0", "dosdevices/com4" );
 }
 
 
