diff -up wine-3.16/dlls/ntdll/server.c.conf~ wine-3.16/dlls/ntdll/server.c
--- wine-3.16/dlls/ntdll/server.c.conf~	2018-09-16 08:54:52.200339498 +0200
+++ wine-3.16/dlls/ntdll/server.c	2018-09-16 09:01:27.541378473 +0200
@@ -1165,6 +1165,12 @@ static int setup_config_dir(void)
     symlink( "../drive_c", "dosdevices/c:" );
     symlink( "/", "dosdevices/z:" );
 
+    /* OpenMandriva specific */
+    symlink( "/media/floppy", "dosdevices/a:" );
+    symlink( getenv( "HOME" ), "dosdevices/d:" );
+    symlink( "/dev/ttyUSB0", "dosdevices/com3" );
+    symlink( "/dev/ttyUSB1", "dosdevices/com4" );
+
 done:
     if (fd_cwd == -1) fd_cwd = open( "dosdevices/c:", O_RDONLY );
     fcntl( fd_cwd, F_SETFD, FD_CLOEXEC );
