diff -up wine-5.8/dlls/ntdll/server.c.conf~ wine-5.8/dlls/ntdll/server.c
--- wine-5.8/dlls/ntdll/server.c.conf~	2020-05-09 17:30:15.873015801 +0200
+++ wine-5.8/dlls/ntdll/server.c	2020-05-09 17:31:25.452856584 +0200
@@ -1593,6 +1593,12 @@ static int setup_config_dir(void)
     }
     else if (errno != EEXIST) fatal_perror( "cannot create %s/dosdevices", config_dir );
 
+    /* OpenMandriva specific */
+    symlink( "/media/floppy", "dosdevices/a:" );
+    symlink( getenv( "HOME" ), "dosdevices/d:" );
+    symlink( "/dev/ttyUSB0", "dosdevices/com3" );
+    symlink( "/dev/ttyUSB1", "dosdevices/com4" );
+
     if (fd_cwd == -1) fd_cwd = open( "dosdevices/c:", O_RDONLY );
     fcntl( fd_cwd, F_SETFD, FD_CLOEXEC );
     return fd_cwd;
