diff -up wine-6.14/configure.ac.omv~ wine-6.14/configure.ac
--- wine-6.14/configure.ac.omv~	2021-08-23 05:13:34.681117692 +0200
+++ wine-6.14/configure.ac	2021-08-23 05:16:45.033781874 +0200
@@ -1528,7 +1528,8 @@ then
          AC_LINK_IFELSE([AC_LANG_PROGRAM([[#ifdef HAVE_NETINET_IN_H
 #include <netinet/in.h>
 #endif
-#include <resolv.h>]],[[if (!(_res.options & RES_INIT)) res_init(); res_query("foo",ns_c_in,0,0,0)]])],
+#include <resolv.h>
+#include <arpa/nameser.h>]],[[if (!(_res.options & RES_INIT)) res_init(); res_query("foo",ns_c_in,0,0,0); ns_initparse(0, 0, 0)]])],
            [ac_cv_have_resolv=${lib:-"none required"}],[ac_cv_have_resolv="not found"])
          test "x$ac_cv_have_resolv" = "xnot found" || break
        done
