diff -up wine-6.21/dlls/capi2032/unixlib.c.omv~ wine-6.21/dlls/capi2032/unixlib.c
--- wine-6.21/dlls/capi2032/unixlib.c.omv~	2021-11-07 13:52:10.520442545 +0100
+++ wine-6.21/dlls/capi2032/unixlib.c	2021-11-07 14:10:28.368352368 +0100
@@ -41,12 +41,21 @@
 #endif
 #include "unixlib.h"
 
+static struct capi20_backend *__cbe = NULL;
+static inline struct capi20_backend *cbe();
+
+static inline struct capi20_backend *cbe() {
+	if(__cbe == NULL)
+		capi20_be_alloc_client(&__cbe);
+	return __cbe;
+}
+
 static NTSTATUS capi_register( void *args )
 {
     struct register_params *params = args;
 
-    return capi20_register( params->maxLogicalConnection, params->maxBDataBlocks, params->maxBDataLen,
-                            (unsigned int *)params->pApplID );
+    return capi20_register( cbe(), params->maxLogicalConnection, params->maxBDataBlocks, params->maxBDataLen,
+                            CAPI_STACK_VERSION, (unsigned int *)params->pApplID );
 }
 
 static NTSTATUS capi_release( void *args )
@@ -82,7 +91,7 @@ static NTSTATUS capi_get_manufacturer( v
 {
     struct get_manufacturer_params *params = args;
 
-    return capi20_get_manufacturer( 0, (unsigned char *)params->SzBuffer ) ? 0 : 0x1108;
+    return capi20_get_manufacturer( cbe(), 0, (char *)params->SzBuffer, sizeof(params->SzBuffer) ) ? 0 : 0x1108;
 }
 
 static NTSTATUS capi_get_version( void *args )
@@ -90,7 +99,7 @@ static NTSTATUS capi_get_version( void *
     struct get_version_params *params = args;
     unsigned int version[4];
 
-    if (!capi20_get_version( 0, (unsigned char *)version )) return 0x1108;
+    if (!capi20_get_version( cbe(), 0, (char *)version, sizeof(version) )) return 0x1108;
     *params->pCAPIMajor         = version[0];
     *params->pCAPIMinor         = version[1];
     *params->pManufacturerMajor = version[2];
@@ -102,19 +111,19 @@ static NTSTATUS capi_get_serial_number(
 {
     struct get_serial_number_params *params = args;
 
-    return capi20_get_serial_number( 0, (unsigned char*)params->SzBuffer ) ? 0 : 0x1108;
+    return capi20_get_serial_number( cbe(), 0, params->SzBuffer, sizeof(params->SzBuffer) ) ? 0 : 0x1108;
 }
 
 static NTSTATUS capi_get_profile( void *args )
 {
     struct get_profile_params *params = args;
 
-    return capi20_get_profile( params->CtlrNr, params->SzBuffer );
+    return capi20_get_profile( cbe(), params->CtlrNr, params->SzBuffer, sizeof(params->SzBuffer) );
 }
 
 static NTSTATUS capi_isinstalled( void *args )
 {
-    return capi20_isinstalled();
+    return capi20_isinstalled(cbe());
 }
 
 unixlib_entry_t __wine_unix_call_funcs[] =
