diff -up wine-5.7/aclocal.m4.omv~ wine-5.7/aclocal.m4
--- wine-5.7/aclocal.m4.omv~	2020-05-06 16:50:10.843775507 +0200
+++ wine-5.7/aclocal.m4	2020-05-06 16:54:52.210898055 +0200
@@ -224,7 +224,7 @@ AC_DEFUN([WINE_CHECK_MINGW_PROG],
   arm*)
     ac_prefix_list="armv7-w64-mingw32-clang armv7-w64-mingw32-gcc" ;;
   i[[3456789]]86*)
-    ac_prefix_list="m4_foreach([ac_wine_prefix],[w64-mingw32, pc-mingw32, mingw32msvc, mingw32],
+    ac_prefix_list="m4_foreach([ac_wine_prefix],[w64-mingw32, w32-mingw32, pc-mingw32, mingw32msvc, mingw32],
                         m4_foreach([ac_wine_cpu],[i686,i586,i486,i386],[ac_wine_cpu-ac_wine_prefix-gcc ]))
                      m4_foreach([ac_wine_cpu],[i686,i586,i486,i386],[ac_wine_cpu-w64-mingw32-clang ])
                      mingw32-gcc" ;;
